base/debian/cuttlefish-base.cuttlefish-host-resources.init

```diff
@@ -136,7 +135,60 @@ destroy_mobile_interface() {
     destroy_tap "${tap}"
 }
 
-# Create many tap devices on a single bridge (wifi or ethernet)
+# Create a wireless tap interface
+# The tap device will have the IP address and DHCP server running on it, and
+# it will be added to a bridge with no address (for routing).
+# $1 = interface number
+# $2 = ip address base ("a.b.c")
+# $3 = IPv6 prefix ("a:b::")
+# $4 = IPv6 prefix length
+create_wireless_interface() {
+    tap="$(printf cvd-wtap-%02d $1)"
+    gateway="${2}.$((4*$1 - 3))"
+    netmask="/30"
+    network="${2}.$((4*$1 - 4))${netmask}"
```

Comment:
Trying to review this I got pretty confused about what `network` is doing here. This is passed in as `-s` to [`iptables(8)`](https://man7.org/linux/man-pages/man8/iptables.8.html) which writes
> [!] -s, --source address[/mask][,...]
>              Source specification. Address can be either a network
>              name, a hostname, a network IP address (with /mask), or a
>              plain IP address. Hostnames will be resolved once only,
>              before the rule is submitted to the kernel.  Please note
>              that specifying any name to be resolved with a remote
>              query such as DNS is a really bad idea.  The mask can be
>              either an ipv4 network mask (for iptables) or a plain
>              number, specifying the number of 1's at the left side of
>              the network mask.  Thus, an iptables mask of 24 is
>              equivalent to 255.255.255.0.
If `network` is a distinct IP address from both the `gateway` and the `dhcp_range`, then what is it actually routing? What packets are produced in with `network` as a source?

If the last bits of `network` don't actually matter because of `netmask` then it could be interesting to make it a `/31` netmask instead and overlap `network` with either `gateway` or `dhcp_range` to be able to pack the ip addresses more tightly. This would address [b/223458270](http://b/223458270) for wifi at least and we might be able to do the same for mobile networking

Can ignore this if you're trying to avoid scope creep though and maybe save the issue for later.

Reply:
I'm not sure if I fully understood your intent, but does your suggestion mean use only 2 IP addresses for each wireless tap device like one for `gateway` and the other for `dhcp_range`? Basically I think it makes sense, but I'm little worried about lack of `broadcast` IP so I need to check that before applying your suggestion.

And about mobile network, would it be enough for just having a ping test after closing network range up?

Reply:
Yeah that is what I meant, sorry about being unclear. Testing `ping` and opening a website in the webview browser should be enough, if possible. Thanks for looking at this!

Reply:
Wi-Fi was worked well when I reduced the range of subnet, but not for mobile network. So, I think supporting 64 CF instances as current would be the best option.

Reply:
Ah unfortunate that it didn't work out, thanks for doing the investigation!